src/zos-io.cc

```diff
@@ -1070,6 +1072,153 @@ bool __doLogMemoryWarning() {
   return __gLogMemoryAll || __gLogMemoryWarning;
 }
 
+// pthread_create override to ensure that _CVTSTATE_OFF does not break multi-threaded programs
+typedef struct {
+    void *(*start_routine)(void *);
+    void *arg;
+} __ThreadArg;
+
+void *custom_start_routine(void *arg) {
+  __ThreadArg *threadArg = (__ThreadArg *)arg;
+
+  int cvstate = __ae_autoconvert_state(_CVTSTATE_QUERY);
+  if (_CVTSTATE_OFF == cvstate) {
+    __ae_autoconvert_state(_CVTSTATE_ON);
+  }
+
+  return threadArg->start_routine(threadArg->arg);
+}
+
+int __pthread_create_extended(pthread_t *thread, const pthread_attr_t *attr,
+                      void *(*start_routine)(void *), void *arg) {
+  __ThreadArg *threadArg = (__ThreadArg *)malloc(sizeof(__ThreadArg));
+  threadArg->start_routine = start_routine;
+  threadArg->arg = arg;
+
+  return __pthread_create_orig(thread, attr, custom_start_routine, (void *)threadArg);
+}
+
+ssize_t getline(char **lineptr, size_t *n, FILE *stream) {
+    if (lineptr == NULL || n == NULL || stream == NULL) {
+        return -1;
+    }
+
+    int c;
+    size_t pos = 0;
+
+    if (*lineptr == NULL || *n == 0) {
+        *n = 128;
```

Comment:
why 128? and why 2 farther down? would those be good to have as buffer length and growth size macros (assuming that's what they are?)